base/show.jl

```diff
@@ -3222,17 +3222,23 @@ end
 summary(io::IO, f::Function) = show(io, MIME"text/plain"(), f)
 
 """
-    showarg(io::IO, x, toplevel)
+    Base.showarg(io::IO, x, toplevel)
 
-Show `x` as if it were an argument to a function. This function is
-used by [`summary`](@ref) to display type information in terms of sequences of
-function calls on objects. `toplevel` is `true` if this is
-the direct call from `summary` and `false` for nested (recursive) calls.
+Show the quasi-type of `x` where quasi-type is the type of `x` or an expression (possibly
+containing quasi-types) that would generate an object of the same type as `x`. The shorter
+of these two options is typically used.
 
-The fallback definition is to print `x` as "::\\\$(typeof(x))",
-representing argument `x` in terms of its type. (The double-colon is
-omitted if `toplevel=true`.) However, you can
-specialize this function for specific types to customize printing.
+This function is used by summary to display type information in terms of sequences of
+function calls on objects.
+
+Show a leading `::` if `toplevel` is `false` and showing a type. `toplevel` is `true` if
+this is the direct call from `summary` and `false` for nested (recursive) calls.
+
+The fallback definition is to print x as "::$(typeof(x))", representing argument x in terms
+of its type. (The double-colon is omitted if toplevel=true.) However, you can specialize
+this function for specific types to customize printing. This customization is useful for
+types that have simple constructors and verbose types such as `reinterpret`ed arrays or
```

Comment:
Might be worth mentioning that this is also useful if the API for the constructor is public but the specifics of the underlying type are not intended to be public facing.

Reply:
Good idea! Done in dff3dfc6f875457ffdfb14a9f0c1c84bf02b4a6c.